### PR TITLE
[core] Reduce self alive check from 60s to 5s.

### PR DESCRIPTION
**Pull request by fishbone**
Signed-off-by: Yi Cheng <74173148+iycheng@users.noreply.github.com>

## Why are these changes needed?

This PR reduce the liveness check from 60s to 5s. It also fixed a bug in the old code where if the server is restarted locally, it'll mark the current one as unhealthy incorrectly because the set is not multi set. 

One example,

- If the head node restart in-place
- when it started, it'll mark the old raylet as dead
- in the liveness check endpoint here, it'll mark the raylet as dead because they share the same ip+port.
- then the head node will exit

Node id should be used for this check in the future for simplicity but the correctness is ok, given: No two raylets can start at the same address (ip+port).

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by iycheng** [comment]
test_serve_agent_fault_tolerane is flakey in master.

**Comment by scv119** [comment]
what problem does reducing the health check interval resolve?

**Comment by iycheng** [comment]
@scv119 this is a revert of revert of https://github.com/ray-project/ray/pull/34769

File: ci/ci.sh [hunk 1]
```diff
@@ -154,7 +154,7 @@ prepare_docker() {
     EXPOSE 8000
     EXPOSE 10001
     RUN pip install /${wheel}[serve]
-    RUN sudo apt update && sudo apt install curl -y
```
Comment (rkooo567) [inline ci/ci.sh:157]: should we revert this change?

**Review by rkooo567 — COMMENTED** [review]
> it'll mark the current one as unhealthy incorrectly because the set is not multi set. 

Can you give me a bit more details? I don't think I understood this part

**Comment by rkooo567** [comment]
@scv119 it is a short term workaround to solve restart problem which @vitsai's PR will fundamentally solve.

File: ci/ci.sh (hunk 1 above)
Comment (iycheng) [reply ci/ci.sh:157]: I think when I run it locally, it failed (`apt update`).
The reason it failed is because some upstream is not responsive.
This is just to make it  more robust. As long as curl can be installed, we shall be good.

**Comment by iycheng** [comment]
@rkooo567 added comments. let me know if you need more explaination.

**Comment by rkooo567** [comment]
> Node id should be used for this check in the future for simplicity but the correctness is ok, given: No two raylets can start at the same address (ip+port).

Why don't we just do this way? If we do it, I believe node_map_ is also unnecessary? (isn't it that we can just check alive_nodes_ instead?)

**Comment by iycheng** [comment]
> > Node id should be used for this check in the future for simplicity but the correctness is ok, given: No two raylets can start at the same address (ip+port).
> 
> Actually, why don't we just do this way? If we do it, I believe node_map_ is also unnecessary? (isn't it that we can just check alive_nodes_ instead?)

That's because for some service, they only have node ip. For example, the node managed by some infra system and they don't have node id and they like to check node's alive.

**Review by rkooo567 — APPROVED** [review]
